.nvidia-ci.yml

```diff
@@ -244,3 +244,69 @@ release:ngc-packaging:
   extends:
     - .dist-packaging
     - .release:ngc
+
+# Define the external image signing steps for NGC
+# Download the ngc cli binary for use in the sign steps
+.ngccli-setup:
+  before_script:
+    - export NGCLI_VERSION=3.41.4
+    - apt-get update && apt-get install -y curl unzip jq
+    - curl -sSLo ngccli_linux.zip https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions/${NGCLI_VERSION}/files/ngccli_linux.zip
+    - unzip ngccli_linux.zip
+    - chmod u+x ngc-cli/ngc
+    - export PATH=$(pwd)/ngc-cli:${PATH}
+    - NGC_CLI_API_KEY=${NGC_REGISTRY_TOKEN} ngc version upgrade --org nvidia -y
+
+# .sign forms the base of the deployment jobs which signs images in the CI registry.
+# This is extended with the image name and version to be deployed.
+.sign:ngc:
+  image: ubuntu:latest
```

Comment:
Question: Does the `ngccli` need Ubuntu, or would the busybox-based image that we use for our other steps be sufficient?

Reply:
ubuntu needed , busybox version will give undefined reference 
* ngc dependency issue due to busybox image . ngc built with latest toolchain and busybox is on older version.
* `ldd $(pwd)/ngc-cli/ngc /lib64/ld-linux-x86-64.so.2 (0x7f55b916c000) libdl.so.2 => /lib64/ld-linux-x86-64.so.2 (0x7f55b916c000) libz.so.1 => /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/libz.so.1 (0x7f55b8f56000) libpthread.so.0 => /lib64/ld-linux-x86-64.so.2 (0x7f55b916c000) libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7f55b916c000) Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/libz.so.1: __vsnprintf_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/libz.so.1: __sprintf_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __strcat_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __snprintf_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __vfprintf_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __realpath_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __strdup: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __memcpy_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __vsnprintf_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __strcpy_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __fread_chk: symbol not found Error relocating /builds/JQq79Cy1/0/dl/container-dev/gpu-operator/ngc-cli/ngc: __fprintf_chk: symbol not found`